.github/workflows/build.yml

```diff
@@ -53,12 +53,51 @@ jobs:
           pip install -U numpy
       - name: Build and install
         run: rm -rf .eggs && pip install -e .
-      - name: Run unittests and generate coverage report
+      - name: Run python unittests and generate coverage report
         run: |
           coverage run --branch --source mmdeploy -m pytest -rsE tests
           coverage xml
           coverage report -m
 
+  build_cpu_sdk:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - name: update
+        run: sudo apt update
+      - name: gcc-multilib
+        run: |
+          sudo apt install gcc-multilib g++-multilib wget libprotobuf-dev protobuf-compiler
+          sudo apt update
+          sudo apt install -y ffmpeg libsm6 libxext6 git ninja-build libglib2.0-0 libxrender-dev libc++1-9 libc++abi1-9
+          sudo add-apt-repository ppa:ignaciovizzo/opencv3-nonfree
+          sudo apt install libopencv-dev lcov wget
+          pkg-config --libs opencv
+      - name: Build and run SDK unit test without backend
+        run: |
+          mkdir -p build && pushd build
+          cmake .. -DCMAKE_CXX_COMPILER=g++-7  -DMMDEPLOY_CODEBASES=all    -DMMDEPLOY_BUILD_SDK=ON     -DMMDEPLOY_BUILD_SDK_PYTHON_API=OFF     -DMMDEPLOY_TARGET_DEVICES=cpu   -DMMDEPLOY_COVERAGE=ON  -DMMDEPLOY_BUILD_TEST=ON
+          make -j2
+          mkdir -p mmdeploy_test_resources/transform
+          wget https://media.githubusercontent.com/media/tpoisonooo/mmdeploy-onnx2ncnn-testdata/main/cat-dog.png
+          cp cat-dog.png mmdeploy_test_resources/transform/
+          ./bin/mmdeploy_tests
+          lcov --capture --directory . --output-file coverage.info
+          ls -lah coverage.info
+          cp coverage.info ../
+      - name: Upload cpp coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          file: ./coverage.info
+          flags: unittests
+          env_vars: OS,CPLUS
+          name: codecov-umbrella
+          fail_ci_if_error: false
```

Comment:
why not let ci fail if something errors happen in the sdk unit tests?

Reply:
`fail_ci_if_error` belongs to `upload codecov` stage.    If unit test fail, CI crash.